sdk/core/azure-core-amqp/vendor/azure-uamqp-c/CMakeLists.txt

```diff
@@ -238,7 +238,7 @@ else()
     )
 endif()
 
-add_library(uamqp
```

Comment:
We're not allowed to touch any files in the azure-uamqp-c directory, because that makes it impossible to commit the changes upstream.

Reply:
I do not see many other ways to do it. Having the cmake as it is creates an export target and installs vendored amqp to the vcpkg directory, potentially overwriting the legit uamqp library that was installed via vcpkg. This has to have no install steps, and I think there are not that many ways around object library so that azure-coreamqp-cpp produces single .lib file with all the symbols.

I made the changes mergeable, and added an extra build flag that our SDK will use. This will not break them.